sdk/core/azure-core/src/credentials/credentials.cpp

```diff
@@ -196,3 +198,70 @@ AccessToken Azure::Core::Credentials::ClientSecretCredential::GetToken(
     throw AuthenticationException("unknown error");
   }
 }
+
+Azure::Core::Credentials::EnvironmentCredential::EnvironmentCredential()
```

Comment:
Should these be in Azure::Core or Azure::Identity?

Reply:
Parts of this *will* become Identity - https://github.com/Azure/azure-sdk-for-cpp/issues/428 I want to do this in a separate PR.

Reply:
Not sure about this specific exception going to Identity or remaining in Core - I'll need to think about it (the very basics DO live in Core in other SDKs too; For example, abstract class TokenCredential lives in Core, while ClientSecretCredential lives in Identity).